packages/core/src/types/errors.ts

```diff
@@ -255,6 +258,17 @@ export interface GroupCycleWarning {
   cycles: string[][];
 }
 
+export interface BBoxApproximationWarning {
+  tag: "BBoxApproximationWarning";
+  // tail is the top of stack
+  stack: [BBoxApproximationWarningItem, ...BBoxApproximationWarningItem[]];
```

Comment:
Similarly, what's the advantage of this type here (vs. `BBoxApproximationWarningItem[]`)?

Reply:
This requires the `BBoxApproximationWarning` to have at least one item in the stack.